tests/integration/consumer/test_consumer_upgrade_downgrade.py

```diff
@@ -0,0 +1,131 @@
+    except Exception:
+        raise
+
+
+def list_offsets(a, topic, no_of_partitions):
+    topic_partition_offsets = {}
+    for partition in range(no_of_partitions):
+        topic_partition = TopicPartition(topic, partition)
+        topic_partition_offsets[topic_partition] = OffsetSpec.latest()
+    futmap = a.list_offsets(topic_partition_offsets, isolation_level=IsolationLevel.READ_COMMITTED, request_timeout=30)
+    for partition, fut in futmap.items():
+        try:
+            result = fut.result()
+            print("Topicname : {} Partition_Index : {} Offset : {} Timestamp : {}"
+                  .format(partition.topic, partition.partition, result.offset,
+                          result.timestamp))
+        except KafkaException as e:
+            print("Topicname : {} Partition_Index : {} Error : {}"
+                  .format(partition.topic, partition.partition, e))
+
+
+# def produce_messages(producer, topic, partitions, num_messages):
+#     for i in range(num_messages):
+#         key = "key-{}".format(i)
+#         value = "value-{}".format(i)
+#         partition = i % partitions
+#         producer.produce(topic, key=key, value=value, partition=partition)
+#     producer.flush()
+
+
+def check_consumer(kafka_cluster, consumers, admin_client, topic, expected_protocol):
+    total_msg_read = 0
+    while len(consumers[-1].assignment()) != number_of_partitions // len(consumers):
+        for consumer in consumers:
+            consumer.poll(0.1)
+
+    for consumer in consumers:
+        assert len(consumer.assignment()) == number_of_partitions // len(consumers)
+
+    assert get_group_protocol_type(admin_client, topic) == expected_protocol
+
+    # Produce some messages to the topic
+    kafka_cluster.seed_topic(topic)
+    list_offsets(admin_client, topic, number_of_partitions)
+
+    while total_msg_read < 100:
+        for consumer in consumers:
+            # Poll for messages
+            msg = consumer.poll(0.1)
+            if msg is not None:
+                total_msg_read += 1
+    
+    assert total_msg_read == 100, "Expected to read 100 messages, but read {}".format(total_msg_read)
+
+
+def perform_consumer_upgrade_downgrade_test_with_partition_assignment_strategy(kafka_cluster, partition_assignment_strategy):
+    """
+    Test consumer upgrade and downgrade.
+    """
+    topic = kafka_cluster.create_topic_and_wait_propogation(topic_prefix,
+                                                                    {
+                                                                        "num_partitions": number_of_partitions
+                                                                    })
+    admin_client = kafka_cluster.admin()
+
+    # Create a consumer with the latest version
+    consumer_conf = {'group.id': topic,
+                     'auto.offset.reset': 'earliest',
+                     'group.protocol': 'classic'}
+    consumer_conf['partition.assignment.strategy'] = partition_assignment_strategy
```

Comment:
Instead of deleting and replacing properties on the dict you could create two dicts
```python
    consumer_conf = {'group.id': topic,
                     'auto.offset.reset': 'earliest'}
    classic_consumer_conf = {
        'group.protocol': 'classic',
        'partition.assignment.strategy': partition_assignment_strategy,
        **consumer_conf
    }
    consumer_consumer_conf = {
        'group.protocol': 'consumer',
        **consumer_conf
    }

```